### PR TITLE
[PC-10508] refacto report modal part 2

### PR DESCRIPTION
**Pull request by yorickeando**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-10508

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Written **documentation on Notion** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/offer/components/__tests__/ReportOffer.native.test.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    const offerId = 0
+
+    const hideReportModal = jest.fn()
+    const navigateToReportReason = jest.fn()
+    const goBackToReportDescription = jest.fn()
+    const navigateToReportOtherReason = jest.fn()
+    const goBackToReportReason = jest.fn()
+
+    it("should switch to report offer reasons upon clicking on 'signaler l'offre' from 1st modal", () => {})
```
Comment (marionnousvalentin) [inline src/features/offer/components/__tests__/ReportOffer.native.test.tsx]: ✅  tu peux enlever ce it là

File: src/features/offer/components/__tests__/ReportOffer.native.test.tsx [hunk 2]
```diff
@@ -159,4 +159,20 @@ describe('ReportOffer', () => {
       fireEvent.press(reportOfferButton)
       expect(hideReportModal).toHaveBeenCalledWith()
     })
+    describe('Report offer description', () => {
+      const ReportOfferFirstModalComponent = render(
```
Comment (marionnousvalentin) [inline src/features/offer/components/__tests__/ReportOffer.native.test.tsx:180]: ✅  tu peux mettre un it(), ça donnera un nom à ton snapshot

File: src/features/offer/components/__tests__/ReportOffer.native.test.tsx [hunk 3]
```diff
@@ -175,4 +175,20 @@ describe('ReportOffer', () => {
       )
       expect(ReportOfferFirstModalComponent).toMatchSnapshot()
     })
+    describe('Report offer reason', () => {
+      const ReportOfferSecondModalComponent = render(
```
Comment (marionnousvalentin) [inline src/features/offer/components/__tests__/ReportOffer.native.test.tsx:196]: ✅ idem ici pour le nom

File: __snapshots__/features/offer/pages/tests/OfferBody.deprecated.native.test.tsx.native-snap [hunk 4]
```diff
@@ -2266,7 +2266,7 @@ Tous les détails du film sur AlloCiné: http://www.allocine.fr/film/fichefilm_g
                                       }
                                     />
```
Comment (marionnousvalentin) [inline __snapshots__/features/offer/pages/tests/OfferBody.deprecated.native.test.tsx.native-snap:2267]: :octocat:  ça, ça va dans le précédent (les changements), pas grave pour cette fois

**Review by marionnousvalentin — APPROVED** [review]
quelques petits changements à faire, mais sinon beau travail !

File: src/features/offer/components/__tests__/ReportOffer.native.test.tsx (hunk 2 above)
Comment (yorickeando) [reply src/features/offer/components/__tests__/ReportOffer.native.test.tsx:180]: top

**Pull request merged**